test/extended/util/framework.go

```diff
@@ -189,7 +189,9 @@ var CheckImageStreamTagNotFoundFn = func(i *imageapi.ImageStream) bool {
 func WaitForADeployment(client kclient.ReplicationControllerInterface,
 	name string,
 	isOK, isFailed func(*kapi.ReplicationController) bool) error {
-	for {
+	startTime := time.Now()
+	endTime := startTime.Add(3 * time.Minute)
```

Comment:
Will do - it worked for me locally but i did not measure how long it
typically took

On Wednesday, December 16, 2015, Ben Parees <notifications@github.com>
wrote:

> In test/extended/util/framework.go
> <https://github.com/openshift/origin/pull/6329#discussion_r47846861>:
>
> > @@ -189,7 +189,9 @@ var CheckImageStreamTagNotFoundFn = func(i *imageapi.ImageStream) bool {
> >  func WaitForADeployment(client kclient.ReplicationControllerInterface,
> >  	name string,
> >  	isOK, isFailed func(*kapi.ReplicationController) bool) error {
> > -	for {
> > +	startTime := time.Now()
> > +	endTime := startTime.Add(3 * time.Minute)
>
> bump this up to something like 15 or 20 minutes. we don't want to
> introduce flakes if we have existing extended tests that take a long time
> waiting for a deployment for some reason.
>
> i'm also surprised 3 minutes is long enough for you test, given that it
> basically has to get through the entire imagebuild+deploy step in 3 minutes
> to pass.
>
> (3 minutes or the watch timeout, whichever is longer.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6329/files#r47846861>.
>